File: 0142-linked-list-cycle-ii/0142-linked-list-cycle-ii.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode detectCycle(ListNode head) {
        HashSet<ListNode> seen = new HashSet();
        ListNode curr = head;
        while(curr!=null){
            if(seen.contains(curr)) return curr;
            seen.add(curr);
            curr=curr.next;
        }
        return null;
    }
}

File: 0142-linked-list-cycle-ii/0142-linked-list-cycle-ii.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var detectCycle = function(head) {
    const seen = new Set();
    let curr = head;
    while(curr!==null){
        if(seen.has(curr)) return curr;
        seen.add(curr);
        curr = curr.next;
    }
    return null;
};